ddtrace/tracer.py

```diff
@@ -296,6 +297,13 @@ def __init__(
         self._shutdown_lock = RLock()
 
         self._new_process = False
+        self._service = config.service
+        config._subscribe(["service"], self._on_global_config_change)
```

Comment:
Is this subscription model a one off change to optimize accessing config.service? Adding subscribers for every configuration could get messy and could be hard to maintain. I don't love this approach but I also can't think of a better alternative